### DISCUSSION TITLE
RTMP YouTube config help

### DISCUSSION THREAD

**Discussion by csb0xc4rs** [gtxaspec/wz_mini_hacks#560] [Q&A] [answered]
I’ve got RTSP working on my V3 running .139 firmware but I cannot sort out why I can’t get RTMP to send a feed to YouTube. I’ve got the appropriate key set etc. Am I missing a simple step? I gave RTMP Enabked of course. Is it because I have not custom defined any bitrate parameters, fps, etc? Just wondering if there is something simple… Or even some debugging I can do? I did notice along the way that in the RTMP script the URL for YouTube starts with a b.* instead of a .* which is which is what YT is expecting on the studio live stream page . Even after modifying that I had no luck. Any thoughts or pointing towards the right debugging or logs are appreciated. Thanks!

**Comment by csb0xc4rs** [comment]
BTW - I did verify RTMP to YT via OBS so I know the key, etc is valid.

**Comment by gtxaspec** [comment]
did you enable RTSP as well? you need both enabled for RTMP to work.

**Reply by csb0xc4rs** [reply]
Yes, RTSP is working and the stream is accessible. I have both streams from RTSP working (High/Low).

**Reply by csb0xc4rs** [reply]
Interim Update (and edit):
I saw an error in the rtmp.log about adding -max_interleave_delta 0 so I did that in rtp-stream.sh as:
VIDEO_OPTIONS="-c:v copy -coder 1 -pix_fmt yuv420p -g 30 -bf 0 -max_interleave_delta 0"

Now I get the stream to display on youtube but am getting complaints about bitrates and resolutions.

I modified VIDEO_OPTIONS one more time to include resolution:
VIDEO_OPTIONS="-c:v copy -coder 1 -pix_fmt yuv420p -g 30 -bf 0 -max_interleave_delta 0 -video_size 1920x1080"

This is at least giving me a full screen live stream on the YT side. Still not sure why I'm only seeing ~800k even though I consistently get 20M up on the home network. Could this just be a limitation of the RTMP capability of the Cam3?

**Reply by csb0xc4rs** [reply, marked as answer]
I think I sorted it all out for RTMP to YouTube. (using Cam3 and 4.36.9.139)

In rtp-streams.sh I needed to modify 2 items:
1. the target url for me is rtmp://a.rtmp..... (not b.rtmp.....)
and
2. VIDEO_OPTIONS="-c:v copy -coder 1 -pix_fmt yuv420p -g 30 -bf 0 -max_interleave_delta 0 -video_size 1920x1080" (added the max_interleave_data and video_size parameters)

In wz_mini.conf I set:
Video0 -> Max (4096) target (2048) enc (2) fps (20)
of course enabled RTMP and added my key
and for what it's worth, I only enabled HI_RES and used 'unicast' as the feed

**Reply by gtxaspec** [reply]
did you have an issue of the video being bordered (small video) on youtube live?  I can't seem to get a full screen.

**Reply by csb0xc4rs** [reply]
Not after forcing the -video_size 1920x1080 parameter in $VIDEO_OPTIONS. I don't know if it is the right way, or most efficient way, but it worked. :)

**Reply by gtxaspec** [reply]
I see.  For some reason it doesn't work for me even with the video size parameters.  I'll keep investigating.

**Reply by csb0xc4rs** [reply]
BTW - Maybe you can see if my config differs from yours. I know I had to do some firmware gyrations to downgrade to .139 for RTSP. I needed to go to .131 first, then to .139. I just updated my Patriot Astro YT channel with a video on this. Thanks for breathing life back into these for us astrophotographers! :) https://www.youtube.com/watch?v=YnkDpAi0buo

**Reply by gtxaspec** [reply]
very cool!

**Reply by Bob-AOL** [reply]
Indeed. Excellent presentation.

**Reply by Bob-AOL** [reply]
Slightly off subject and you may or may not find this useful. I have no idea what your setup may entail, but this seems like it could possibly enhance your image results. There are also various lens options that are easily and economically swapped in and out once you've taken a v3 apart a time or two. 

https://www.youtube.com/watch?v=kkKAMMNlaf4&t=38s

**Reply by csb0xc4rs** [reply]
ooooo.... I do have a couple watching my 3d printers. I think I will be ripping apart some cameras in 3...2...1... :)

**Reply by csb0xc4rs** [reply]
LOL, Ok, so I am now in focus at a shorter distance. Thanks! But, ... I watched the video and I swear all I must have heard was "it takes 20 minutes" because I worked on it like I was in a race thinking it would be quick. Well, it was quick, but not without me setting myself back more than one time. :) All in all, 2.5 hours but that was with me designing and printing the focus ring as well (PETG since I had that on hand). Thanks again!

**Reply by Bob-AOL** [reply]
LOL Glad it worked out. Seems pretty efficient to me! You now see how easy it is to experiment with various lenses as well.

So have you taken the USB/POE/ETH plunge yet? That's the real game changer with wz_mini.

**Reply by csb0xc4rs** [reply]
the usb-enet just showed up at my front door :) I think I have some lenses kicking around somehwere from and old allsky camera project with a RPI4 camera. May have to look around.

**Reply by Bob-AOL** [reply]
Very good. If you got the UCTRONICS adapter and experience any issues, assistance is just a click away.
